lib/log_hog.ex

```diff
@@ -1,2 +1,201 @@
 defmodule LogHog do
+  @typedoc "Name under which an instance of LogHog supervision tree is registered."
+  @type supervisor_name() :: atom()
+
+  @typedoc "Event name, such as `\"user_signed_up\"` or `\"$create_alias\"`"
+  @type event() :: String.t()
+
+  @typedoc "string representing distinct ID"
+  @type distinct_id() :: String.t()
+
+  @typedoc """
+  Map representing event properties.
+
+  Note that it __must__ be JSON-serializable.
+  """
+  @type properties() :: %{optional(String.t()) => any(), optional(atom()) => any()}
```

Comment:
Can we guarantee JSON serialization with types? Maybe we should be explicit and require `atom() | String.t() | number | true | false | Map.t() | List.t()`? (And any extra JSON-serializable property I'm now forgetting)

Reply:
I don't think we can guarantee anything with typespecs. At best they might warn users if they run dialyzer. I'm not sure if there is a way to express json-seiralizable data as type, as in the end it will boil down to what encoder is used and if user's data implements needed protocols. Users may pass structs here and make them serialize into strings and whatnot! The official `JSON.encode!` [typespec](https://hexdocs.pm/elixir/JSON.html#encode!/2) uses `term()` as input type.